crates/services/p2p/src/codecs/postcard.rs

```diff
@@ -85,28 +82,36 @@ impl RequestResponseCodec for PostcardCodec {
 
     async fn read_request<T>(
         &mut self,
-        _protocol: &Self::Protocol,
+        _: &Self::Protocol,
         socket: &mut T,
     ) -> io::Result<Self::Request>
     where
         T: AsyncRead + Unpin + Send,
     {
-        let encoded_data = read_length_prefixed(socket, MAX_REQUEST_SIZE).await?;
+        let mut response = Vec::new();
+        socket
+            .take(self.max_response_size as u64)
+            .read_to_end(&mut response)
+            .await?;
 
-        self.deserialize(&encoded_data)
+        self.deserialize(&response)
```

Comment:
todo: we may want to look into streaming deserialization at some point instead of buffering the whole message into a vec first.

Reply:
https://github.com/FuelLabs/fuel-core/issues/1563